setup typos check in CI

klensy opened the pull request:
This allows to check typos in CI, currently for compiler only (to reduce commit size with fixes). With current setup, exclude list is quite short, so it worth trying?

Also includes commits with actual typo fixes.

MCP: https://github.com/rust-lang/compiler-team/issues/817

typos check currently turned for:
* ./compiler
* ./library
* ./src/bootstrap
* ./src/librustdoc

After merging, PRs which enables checks for other crates (tools) can be implemented too.

Found typos will **not break** other jobs immediately: (tests, building compiler for perf run). Job will be marked as red on completion in ~ 20 secs, so you will not forget to fix it whenever you want, before merging pr.

Check typos: `python x.py test tidy --extra-checks=spellcheck`
Apply typo fixes: `python x.py test tidy --extra-checks=spellcheck:fix` (in case if there only 1 suggestion of each typo)

Current fail in this pr is expected and shows how typo errors emitted. Commit with error will be removed after r+.

klensy committed:
setup CI and tidy to use typos for spellchecking and fix few typos

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12212695377/job/34071454317) [(plain)](https://github.com/rust-lang/rust/commit/97b7b449e9e8b229804d9517b9deb326e0c9c157/checks/34071454317/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
Download action repository 'crate-ci/typos@v1.28.2' (SHA:2872c382bb9668d4baa5eade234dcbc0048ca2cf)
Download action repository 'actions/upload-artifact@v4' (SHA:b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882)
Complete job name: PR - mingw-check
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
---
   Compiling basic-toml v0.1.9
   Compiling rinja_parser v0.3.5
   Compiling rinja_derive v0.3.5
    Checking rinja v0.3.5
error[E0599]: no method named `prefer_remapped_unconditionaly` found for enum `FileName` in the current scope
    |
    |
750 |             format!("{} - {item_path}(line {line})", filename.prefer_remapped_unconditionaly());
    |
help: there is a method `prefer_remapped_unconditionally` with a similar name
    |
    |
750 |             format!("{} - {item_path}(line {line})", filename.prefer_remapped_unconditionally());

For more information about this error, try `rustc --explain E0599`.
error: could not compile `rustdoc` (lib) due to 1 previous error
Build completed unsuccessfully in 0:02:35
```

</details>

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12212821750/job/34071747656) [(plain)](https://github.com/rust-lang/rust/commit/5b7daa73243739a876d7bdf44d5a3dba18c8c74e/checks/34071747656/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
Download action repository 'crate-ci/typos@v1.28.2' (SHA:2872c382bb9668d4baa5eade234dcbc0048ca2cf)
Download action repository 'actions/upload-artifact@v4' (SHA:b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882)
Complete job name: PR - mingw-check
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
---
    Checking rustfix v0.8.1
    Checking colored v2.1.0
    Checking tempfile v3.13.0
    Checking ui_test v0.26.5
error[E0599]: no method named `prefer_remapped_unconditionaly` found for enum `FileName` in the current scope
    |
    |
133 |         let filename = lo.file.name.prefer_remapped_unconditionaly().to_string();
    |
help: there is a method `prefer_remapped_unconditionally` with a similar name
    |
    |
133 |         let filename = lo.file.name.prefer_remapped_unconditionally().to_string();

For more information about this error, try `rustc --explain E0599`.
error: could not compile `miri` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

</details>

Nadrieril commented on compiler/rustc_parse/src/parser/diagnostics.rs:
```diff
@@ -795,7 +795,7 @@ impl<'a> Parser<'a> {
         // token is an identifier, it's probably a misspelled keyword.
         // This handles code like `async Move {}`, misspelled `if` in match guard, misspelled `else` in `if`-`else`
-        // and mispelled `where` in a where clause.
+        // and misspelled `where` in a where clause.
```
they misspelled "misspelled" :grinning:

Nadrieril commented on compiler/rustc_pattern_analysis/src/usefulness.rs:
```diff
@@ -102,7 +102,7 @@
 //!
 //! Note 1: variable bindings (like the `x` in `Some(x)`) match anything, so we treat them as wildcards.
-//! Note 2: this only applies to matcheable values. For example a value of type `Rc<u64>` can't be
+//! Note 2: this only applies to matchable values. For example a value of type `Rc<u64>` can't be
```
Didn't you add an exception for this in `typos.toml`? I don't know which is correct but tf there's an exception this should be kept, presumably.

klensy replied:
reverted

GuillaumeGomez replied:
Doesn't seem to be reverted. ;)

klensy replied:
Here https://github.com/rust-lang/rust/pull/134006#discussion_r1875050909 reviewer said that this is not valid word, so i removed exception somewhere here https://github.com/rust-lang/rust/pull/134006/commits/199e4e73128d6ee88510a106dfa7c0ad313bf953.

Nadrieril commented on typos.toml:
```diff
@@ -0,0 +1,41 @@
+EXCED = "EXCED"
+Shortern = "Shortern"
+unparseable = "unparseable"
+replacable = "replacable"
```
Is that not a typo of "replaceable"?

klensy replied:
Internet says that correct variant is `replaceable` (even my spellchecker in browser), but `replacable` still used. Will fix then.

Nadrieril commented on typos.toml:
```diff
@@ -0,0 +1,41 @@
+[files]
+extend-exclude = [
+    # ./compiler excludes
+    "compiler/rustc_codegen_gcc",
+    "compiler/rustc_codegen_cranelift",
+    "compiler/rustc_baked_icu_data",
+]
```
Could you add a comment explaining why we exclude these?

klensy replied:
added

RalfJung commented on typos.toml:
```diff
@@ -0,0 +1,41 @@
+    "compiler/rustc_baked_icu_data",
+]
+
+[default.extend-words]
```
What is the meaning of this section? All of the entries here are `x = "x"`, so this doesn't seem to *extend* anything? Would be worth a comment so the file can be more easily edited by people that are not familiar with the underlying tooling.

klensy replied:
Added description and link to docs.

Nadrieril commented:
I seem to recall a previous attempt at having typo-checks in CI that had a ton of false positives, this is much nicer, thanks. I can't approve the CI addition myself (does this require an FCP maybe?), but I looked through the typo fixes and they all look good to me.

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12212934507/job/34072000264) [(plain)](https://github.com/rust-lang/rust/commit/5290542dbd3c0fde0c34a26fc32f5fbabd6a1dfa/checks/34072000264/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
Download action repository 'crate-ci/typos@v1.28.2' (SHA:2872c382bb9668d4baa5eade234dcbc0048ca2cf)
Download action repository 'actions/upload-artifact@v4' (SHA:b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882)
Complete job name: PR - mingw-check
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
---


# RECOMMENDATIONS

* Fix missing copyright/licensing information: For one or more files, the tool
  cannot find copyright and/or licensing information. You typically do this by
  adding 'SPDX-FileCopyrightText' and 'SPDX-License-Identifier' tags to each
  file. The tutorial explains additional ways to do this:
  <https://reuse.software/tutorial/>
  local time: Sat Dec  7 13:07:51 UTC 2024
  network time: Sat, 07 Dec 2024 13:07:52 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

lqd commented on typos.toml:
```diff
@@ -0,0 +1,44 @@
+optin = "optin"
+parm = "parm"
+EXCED = "EXCED"
+Shortern = "Shortern"
```
This shouldn’t be excluded

lqd commented:
> does this require an FCP maybe?

maybe an MCP?

the8472 commented on typos.toml:
```diff
@@ -0,0 +1,42 @@
+]
+
+[default.extend-words]
+# Add exclusions here, lines should be like `x = "x"`, where `x` is excluded word.
+#
+# Also see docs: https://github.com/crate-ci/typos/blob/v1.28.2/docs/reference.md
```
Based on the documentation these aren't exclusions, they're custom additions to the dictionary.

klensy replied:
Yes, but it is one of the ways to silence linter, so i wrote as `exclusions`, to simplify wording.

Nadrieril commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+makro = "makro"
+optin = "optin"
+parm = "parm"
+EXCED = "EXCED"
```
That typo only shows up here, may as well include just that identifier.
```suggestion
ERROR_FILENAME_EXCED_RANGE = "ERROR_FILENAME_EXCED_RANGE"
```

klensy replied:
Bad rebase, but still applied fix.

saethlin commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+]
+
+extend-ignore-re = [
+    # ignore this intensional typo examples
```
```suggestion
    # ignore these intentional typo examples
```

Nadrieril commented:
On second thought, should this be included in `./x.py test tidy` instead? That would make it possible to auto-apply with `./x.py test tidy --bless`.

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12216025155/job/34078696478) [(plain)](https://github.com/rust-lang/rust/commit/e7baf2e6d0bfe6b3c121b9f0b30b84a83db24399/checks/34078696478/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
Download action repository 'crate-ci/typos@v1.28.2' (SHA:2872c382bb9668d4baa5eade234dcbc0048ca2cf)
Download action repository 'actions/upload-artifact@v4' (SHA:b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882)
Complete job name: PR - mingw-check-tidy
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
---
  INPUT_ISOLATED: false
  INPUT_WRITE_CHANGES: false
  INPUT_CONFIG: ./typos.toml
##[endgroup]
Downloading 'typos' v1.28.2
--2024-12-07 20:54:38--  https://github.com/crate-ci/typos/releases/download/v1.28.2/typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/181782286/a085a802-9666-47b6-9804-a6c2de9b06c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241207%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241207T205438Z&X-Amz-Expires=300&X-Amz-Signature=82f2d571c544d9545c3e29e23d2f7b44b166b6114df04bfd438117b7c925aafb&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dtypos-v1.28.2-x86_64-unknown-linux-musl.tar.gz&response-content-type=application%2Foctet-stream [following]
--2024-12-07 20:54:38--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/181782286/a085a802-9666-47b6-9804-a6c2de9b06c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241207%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241207T205438Z&X-Amz-Expires=300&X-Amz-Signature=82f2d571c544d9545c3e29e23d2f7b44b166b6114df04bfd438117b7c925aafb&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dtypos-v1.28.2-x86_64-unknown-linux-musl.tar.gz&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.109.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.111.133|:443... connected.
Length: 4266444 (4.1M) [application/octet-stream]
Length: 4266444 (4.1M) [application/octet-stream]
Saving to: ‘typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz’
     0K ........ ........ ........ ........ ........ ........ 73%  253M 0s
  3072K ........ ........ .                                  100%  340M=0.02s


2024-12-07 20:54:38 (271 MB/s) - ‘typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz’ saved [4266444/4266444]
./typos
jq: jq-1.6
jq: jq-1.6
$ ./typos ./compiler ./library --config ./typos.toml
##[warning]"outter" should be "outer".
##[warning]"outter" should be "outer".
error: `outter` should be `outer`
  --> ./compiler/rustc_ast/src/ast.rs:1204:58
     |
1204 |     /// Returns an expression with (when possible) *one* outter outter brace removed
     |
     |
error: `outter` should be `outer`
  --> ./compiler/rustc_ast/src/ast.rs:1204:65
     |
1204 |     /// Returns an expression with (when possible) *one* outter outter brace removed
     |
##[error]Process completed with exit code 2.
##[group]Run jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be
with:
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.674 Building wheels for collected packages: reuse
#13 2.675   Building wheel for reuse (pyproject.toml): started
#13 2.921   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 2.922   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 2.922   Stored in directory: /tmp/pip-ephem-wheel-cache-66pvrp4x/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 2.925 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.311 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.311 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.832 Collecting virtualenv
#13 3.832 Collecting virtualenv
#13 3.870   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 3.935      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 68.0 MB/s eta 0:00:00
#13 3.974 Collecting distlib<1,>=0.3.7
#13 3.978   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 3.985      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 88.6 MB/s eta 0:00:00
#13 4.020 Collecting filelock<4,>=3.12.2
#13 4.023   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.056 Collecting platformdirs<5,>=3.9.1
#13 4.059   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.143 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.322 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.4s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      321472 kB
DirectMap2M:     7018496 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
fmt: skipped untracked file typos
fmt: checked 5701 files
tidy check
tidy: Skipping binary file check, read-only filesystem
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (24.3.1)
All checks passed!
checking C++ file formatting
found untracked files, ignoring
x.py completions check
```

</details>

klensy commented:
> On second thought, should this be included in `./x.py test tidy` instead? That would make it possible to auto-apply with `./x.py test tidy --bless`.

It it possible, but will force installing linter for all users of tidy, which isn't nice. Or it can be implemented as external linter for tidy, like `ruff` or `clang-format` currently. Anyway, i don't expect spellchecker emit *so* many warnings that it will require automatic bless.

Nadrieril commented:
It's not so much about the amount of warnings, it's about the annoying fact of having to wait 10 minutes for CI only to realize there was a typo. External linter is fine.

klensy commented:
Example of running spellchecker:
![image](https://github.com/user-attachments/assets/4d83ad47-ddb0-4a60-a503-1c7a1298a1c3)
All jobs competed even when there is some typos, so it will not force fixing typos before running other tests.

Or spellcheker can be moved to separate independent job, so it will give results in ~ 20 seconds.

klensy commented:
>wait 10 minutes for CI only to realize there was a typo

Yes, but i personally don't want to fix pr twice anyway, in case if there was typo and some other tidy issues, like formatting, better see them in one chunk.

Nadrieril commented:
But then I fix the typo, push a commit, and have to wait for CI again. It interrupts the flow, I might have to change back to the correct branch if I went to work on another one etc. I'm not saying it's horrible as-is, just that having it in `tidy` would make our lives easier. Especially since the MCP has a "contributor friction" concern right now.

Nadrieril commented:
> or some other tidy issues, like formatting; better see them in one chunk.

That's why we recommend `tidy` in a pre-push hook, so you never have CI fail for such trivial reasons.

klensy commented:
Pushed change: it's possible to check typos via `python x.py test tidy --extra-checks=spellcheck` or bless via `python x.py test tidy --extra-checks=spellcheck:fix`

rust-log-analyzer commented:
The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12221693934/job/34091082029) [(plain)](https://github.com/rust-lang/rust/commit/1ad9b304552d20e0dcc4271e2cf4e284f142d816/checks/34091082029/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b1630ceea73d27597364c9af683)
Download action repository 'crate-ci/typos@v1.28.2' (SHA:2872c382bb9668d4baa5eade234dcbc0048ca2cf)
Download action repository 'actions/upload-artifact@v4' (SHA:b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882)
Complete job name: PR - mingw-check-tidy
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
---
  INPUT_ISOLATED: false
  INPUT_WRITE_CHANGES: false
  INPUT_CONFIG: ./typos.toml
##[endgroup]
Downloading 'typos' v1.28.2
--2024-12-08 12:17:03--  https://github.com/crate-ci/typos/releases/download/v1.28.2/typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz
Resolving github.com (github.com)... 140.82.112.3
Connecting to github.com (github.com)|140.82.112.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/181782286/a085a802-9666-47b6-9804-a6c2de9b06c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241208T121703Z&X-Amz-Expires=300&X-Amz-Signature=d5f6187add83a34482e585ee4766f670f1dae63ab5dd1cb3ddb165cfe0074008&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dtypos-v1.28.2-x86_64-unknown-linux-musl.tar.gz&response-content-type=application%2Foctet-stream [following]
--2024-12-08 12:17:03--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/181782286/a085a802-9666-47b6-9804-a6c2de9b06c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20241208%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20241208T121703Z&X-Amz-Expires=300&X-Amz-Signature=d5f6187add83a34482e585ee4766f670f1dae63ab5dd1cb3ddb165cfe0074008&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dtypos-v1.28.2-x86_64-unknown-linux-musl.tar.gz&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.
Length: 4266444 (4.1M) [application/octet-stream]
Length: 4266444 (4.1M) [application/octet-stream]
Saving to: ‘typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz’
     0K ........ ........ ........ ........ ........ ........ 73%  242M 0s
  3072K ........ ........ .                                  100%  356M=0.02s


2024-12-08 12:17:03 (265 MB/s) - ‘typos-v1.28.2-x86_64-unknown-linux-musl.tar.gz’ saved [4266444/4266444]
./typos
jq: jq-1.6
jq: jq-1.6
$ ./typos ./compiler ./library --config ./typos.toml
##[warning]"outter" should be "outer".
##[warning]"outter" should be "outer".
error: `outter` should be `outer`
  --> ./compiler/rustc_ast/src/ast.rs:1204:58
     |
1204 |     /// Returns an expression with (when possible) *one* outter outter brace removed
     |
     |
error: `outter` should be `outer`
  --> ./compiler/rustc_ast/src/ast.rs:1204:65
     |
1204 |     /// Returns an expression with (when possible) *one* outter outter brace removed
     |
##[error]Process completed with exit code 2.
##[group]Run jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be
with:
---

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.780 Building wheels for collected packages: reuse
#13 2.782   Building wheel for reuse (pyproject.toml): started
#13 3.024   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.025   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.026   Stored in directory: /tmp/pip-ephem-wheel-cache-05_l6mhv/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.028 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.418 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.418 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.943 Collecting virtualenv
#13 3.943 Collecting virtualenv
#13 3.979   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.043      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 68.4 MB/s eta 0:00:00
#13 4.082 Collecting distlib<1,>=0.3.7
#13 4.088   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.096      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 84.6 MB/s eta 0:00:00
#13 4.129 Collecting filelock<4,>=3.12.2
#13 4.132   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.164 Collecting platformdirs<5,>=3.9.1
#13 4.168   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.250 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.430 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      325568 kB
DirectMap2M:     9111552 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
downloading https://static.rust-lang.org/dist/2024-11-27/rustc-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-11-27/rustc-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
x.py completions were changed; run `x.py run generate-completions` to update them
  local time: Sun Dec  8 12:27:05 UTC 2024
  network time: Sun, 08 Dec 2024 12:27:05 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

onur-ozkan commented on .github/workflows/ci.yml:
```diff
@@ -58,6 +58,32 @@ jobs:
+      - name: check typos
+        # sync version with src/tools/tidy/src/ext_tool_checks.rs
+        uses: crate-ci/typos@v1.28.2
+        with:
+        # sync target files with src/tools/tidy/src/ext_tool_checks.rs
+          files: ./compiler ./library
```
```suggestion
        with:
          # sync target files with src/tools/tidy/src/ext_tool_checks.rs
          files: ./compiler ./library
```

nnethercote commented on compiler/rustc_errors/src/tests.rs:
```diff
@@ -97,7 +97,7 @@ fn wellformed_fluent() {
 
 #[test]
-fn misformed_fluent() {
+fn malformed_fluent() {
```
"misformed" is a word

klensy replied:
fixed

nnethercote commented on compiler/rustc_codegen_ssa/src/back/metadata.rs:
```diff
@@ -408,7 +408,7 @@ pub(crate) fn create_object_file(sess: &Session) -> Option<write::Object<'static
 /// - The platform/OS they were built for (macOS/watchOS/Mac Catalyst/iOS simulator etc).
 /// - The minimum OS version / deployment target.
-/// - The version of the SDK they were targetting.
+/// - The version of the SDK they were targeting.
```
"targetting" appears to be an acceptable British English spelling

nnethercote commented on compiler/rustc_lint_defs/src/builtin.rs:
```diff
@@ -4442,7 +4442,7 @@ declare_lint! {
 
 declare_lint! {
     /// The `refining_impl_trait_reachable` lint detects `impl Trait` return
-    /// types in method signatures that are refined by a publically reachable
```
"publically" is less common than "publicly", but does get some use

klensy replied:
fixed

nnethercote commented on library/core/src/primitive_docs.rs:
```diff
@@ -1628,8 +1628,8 @@ mod prim_ref {}
 /// assert_eq!(ptr(5), 6);
 ///
-/// let clos: fn(usize) -> usize = |x| x + 5;
-/// assert_eq!(clos(5), 10);
```
I don't love having to change this, abbreviations get used for identifiers a lot.

klensy replied:
One small public docstring with example.

nnethercote replied:
Not checking four letter words would avoid needing this change.

nnethercote commented on library/std/src/sys_common/mod.rs:
```diff
@@ -68,17 +68,17 @@ pub trait FromInner<Inner> {
     fn from_inner(inner: Inner) -> Self;
 }
 
-// Computes (value*numer)/denom without overflow, as long as both
```
Ugh, another case of having to change a reasonable identifier.

klensy replied:
Okay.

nnethercote commented on src/tools/tidy/src/ext_tool_checks.rs:
```diff
@@ -241,6 +243,22 @@ fn check_impl(
+    if spellcheck_all || spellcheck_fix {
+        let config_path = root_path.join("typos.toml");
+        let mut args =
+        // sync target files with .github/workflows/ci.yml
```
Comment in the middle of the assignment is ugly, move it above.

nnethercote commented on src/tools/tidy/src/ext_tool_checks.rs:
```diff
@@ -241,6 +243,22 @@ fn check_impl(
+
+        if spellcheck_all {
+            eprintln!("spellcheck files");
+            spellcheck_runner(&args)?;
```
You can factor out the duplicated `spellcheck_runner(&args)?;`.

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+optin = "optin"
+parm = "parm"
+unparseable = "unparseable"
+matcheable = "matcheable"
```
Why is this here? Not a valid word, AFAIK

klensy replied:
ok.

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+childs = "childs"
+filetimes = "filetimes"
+
+[default.extend-identifiers]
```
Feels like we need comments explaining these exceptions.

klensy replied:
Added more words for description

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+unstalled = "unstalled"
+taits = "taits"
+Datas = "Datas"
+splitted = "splitted"
```
I'd probably not allow "splitted", "split" is preferable.

klensy replied:
Valid too https://www.oed.com/search/dictionary/?scope=Entries&q=splitted, but rare?

nnethercote replied:
hmm, ok

nnethercote commented on library/core/src/intrinsics/simd.rs:
```diff
@@ -85,7 +85,7 @@ extern "rust-intrinsic" {
     pub fn simd_shr<T>(lhs: T, rhs: T) -> T;
 
-    /// "Ands" vectors elementwise.
+    /// "And"s vectors elementwise.
```
This one is a bit dubious.

klensy replied:
`And's`?

nnethercote replied:
ok

nnethercote commented on library/std/src/sys/pal/unix/time.rs:
```diff
@@ -261,8 +261,8 @@ impl Instant {
         //
         // CLOCK_UPTIME_RAW   clock that increments monotonically, in the same man-
-        //                    ner as CLOCK_MONOTONIC_RAW, but that does not incre-
-        //                    ment while the system is asleep.  The returned value
```
ugh

nnethercote replied:
Not checking 4 letter words would avoid needing this change.

klensy replied:
ok.

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+# Add exclusions here, lines should be like `x = "x"`, where `x` is excluded word.
+#
+# Also see docs: https://github.com/crate-ci/typos/blob/v1.28.2/docs/reference.md
+thir = "thir"
```
What's the format of this file? Having to repeat the word on every line is annoying.

klensy replied:
Format described in linked docs: https://github.com/crate-ci/typos/blob/v1.28.2/docs/reference.md

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,62 @@
+
+[default]
+extend-ignore-words-re = [
+    # words with length <= 3 chars is likely noise
```
I would increase this from 3 to 4. Misspelled four letter words are very rare. The only examples above are all false positives: "ands", "clos" (short for "closure"), and "ment" (in the split "incre-ment").

nnethercote replied:
I still think this is a good idea.

klensy replied:
kk.

nnethercote commented:
From a code POV I think this is on track to be acceptable. Seems like there is some disagreements in the MCP, though, so that's likely to be the primary hurdle.

klensy commented:
At time of MCP I had an idea, but reality of github CI action implementation prevented some nicer part of it, so i will revert to simple implementation and will resolve review comments.

klensy commented:
Oops, weird rebase.

nnethercote commented on compiler/rustc_lint_defs/src/builtin.rs:
```diff
@@ -4484,7 +4484,7 @@ declare_lint! {
     /// `refining_impl_trait` is a lint group composed of two lints:
     ///
-    /// * `refining_impl_trait_reachable`, for refinements that are publically
+    /// * `refining_impl_trait_reachable`, for refinements that are publicly
```
doesn't need changing; likewise the one just below

klensy replied:
done.

nnethercote commented on typos.toml:
```diff
@@ -0,0 +1,73 @@
+publically = "publically"
+
+[default.extend-identifiers]
+# Entries goes here if typo is part of some existing ident
```
```suggestion
# An entry goes here if the typo is part of some existing ident
```

bors commented:
:umbrella: The latest upstream changes (presumably #133793) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

GuillaumeGomez commented on compiler/rustc_ast/src/ast.rs:
```diff
@@ -1194,7 +1194,7 @@ impl Expr {
     }
 
-    /// Returns an expression with (when possible) *one* outter brace removed
+    /// Returns an expression with (when possible) *one* outter outter brace removed
```
Is it for testing CI or just a typo?

klensy replied:
This is testing typo from `[dont merge] add typo` commit.

GuillaumeGomez replied:
Perfect, resolving discussion then, thanks!

bors commented:
:umbrella: The latest upstream changes (presumably #134286) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:umbrella: The latest upstream changes (presumably #135896) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:umbrella: The latest upstream changes (presumably #136549) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:umbrella: The latest upstream changes (presumably #136728) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

bors commented:
:umbrella: The latest upstream changes (presumably #139746) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

nnethercote commented on compiler/rustc_hir/src/hir.rs:
```diff
@@ -4224,7 +4224,7 @@ impl fmt::Display for Constness {
 }
 
-/// The actualy safety specified in syntax. We may treat
+/// The actually safety specified in syntax. We may treat
```
I think this should be "actual"? That makes more sense than "actually", though I'm not certain.

@oli-obk: git tells me you wrote this comment, can you confirm?

klensy replied:
done

nnethercote commented on src/tools/tidy/src/ext_tool_checks.rs:
```diff
@@ -491,6 +514,25 @@ fn shellcheck_runner(args: &[&OsStr]) -> Result<(), Error> {
+                "typos",
+                "spellcheck file checks",
+                // sync version with .github/workflows/ci.yml
+                Some("install tool via `cargo install typos-cli@1.34.0`".to_owned()),
```
Does it have to be v1.34.0? Would it make more sense to omit the version number? Seems like "the latest version" would be better.

nnethercote replied:
And s/ci.yml/spellcheck.yml/, as above.

klensy replied:
I've added commit with check that ensures that correct version is installed. In case if version differs, output from local check and CI will diverge (which is bad). This should probably be not fatal error in case if people have different version for whatever case.

nnethercote commented on src/tools/tidy/src/ext_tool_checks.rs:
```diff
@@ -224,6 +226,27 @@ fn check_impl(
 
+    if spellcheck_all || spellcheck_fix {
+        let config_path = root_path.join("typos.toml");
+        // sync target files with .github/workflows/ci.yml
```
s/ci.yml/spellchecks.yml/

nnethercote commented:
Thanks for your patience on this, I think it's very close to being ready. That's a lot of nice typo fixes.
- Please address my comments above.
- The "don't merge" commit with the "outter" typo can now be removed.
- Please squash all the other commits; I think this one is simpler to understand as a single commit.

And just to confirm my understanding:
- This will run by default on CI, as a separate "Spellcheck" check. (Seen to deliberately fail above.)
- It won't run in local `x test tidy` runs by default, because people might not have typos-cli installed.
- You can run it locally with `x test tidy --extra-checks=spellcheck` if you have typos-cli installed.

klensy commented:
Squashed previous commits and added review commit + commit where required version of typos checked.

> This will run by default on CI, as a separate "Spellcheck" check. (Seen to deliberately fail above.)

Yes, run by default by separate job and will not immediately stop any other jobs in case if spellcheck fails.

> It won't run in local x test tidy runs by default, because people might not have typos-cli installed.

Nope, running `x test tidy` without `--extra-checks=spellcheck` or `--extra-checks=spellcheck:fix` will not call spellchecker.

> You can run it locally with x test tidy --extra-checks=spellcheck if you have typos-cli installed.

Yes, and and can fix typos if `x test tidy --extra-checks=spellcheck:fix` used (if there no multiple suggestions).

nnethercote commented:
Looks good! Squash the final two commits into the first commit and r=me. Thanks.

@bors delegate=klensy

bors commented:
:pushpin: Commit c76d032f0144b650a438ee1efba89c475e0b115b has been approved by `nnethercote`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=nnethercote c76d032f0144b650a438ee1efba89c475e0b115b -->
<!-- homu: {"type":"Approved","sha":"c76d032f0144b650a438ee1efba89c475e0b115b","approver":"nnethercote","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Noratrieb commented:
FYI: This PR integrated typo checking into CI incorrectly by merely adding a new workflow job instead of properly integrating it into our merge CI.

This caused the tree to break ([#t-infra > Spellcheck workflow now fails on all PRs (tree bad?)](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/Spellcheck.20workflow.20now.20fails.20on.20all.20PRs.20.28tree.20bad.3F.29/with/529580411)) because a PR with a typo was merged after this PR was merged. This is something our CI is designed to prevent via the no rocket science rule, but that only works when the jobs run on merge CI as intended.

Any changes to CI should always go through the infra team to help prevent such mistakes.

nnethercote commented:
Sorry for the bustage! Judging by [this comment](https://rust-lang.zulipchat.com/#narrow/channel/242791-t-infra/topic/Spellcheck.20workflow.20now.20fails.20on.20all.20PRs.20.28tree.20bad.3F.29/near/529598625) there was some infra team awareness, and the sub-optimal config was similar to how tidy had worked in the past, which explains why it wasn't flagged as a problem. I.e. the situation was a bit complicated.

nnethercote commented:
Sounds like work is underway to tighten things up, so maybe we'll have a happy outcome :)
